rust/gitxetcore/src/command/materialize.rs

```diff
@@ -56,11 +77,37 @@ pub async fn materialize_command(cfg: XetConfig, args: &MaterializeArgs) -> Resu
     drop(lazyconfig);
 
     // rerun smudge filter
-    std::fs::remove_file(cfg.repo_path()?.join("index"))?;
+    let mut git_args = vec!["--"];
+
+    // touch all files we want to materialize
+    for p in path_list_ref.iter() {
+        let relative_path_from_curdir =
+            Path::new(p)
+                .strip_prefix(relative_to_workroot)
+                .map_err(|e| {
+                    error!(
+                        "Failed to get relative path for {p:?} to the current working directory."
+                    );
+                    GitXetRepoError::Other(e.to_string())
+                })?;
+        touch_file(relative_path_from_curdir)?;
+        git_args.push(relative_path_from_curdir.to_str().unwrap_or_default());
+    }
 
-    git_wrap::run_git_captured(None, "checkout", &["--force"], true, None)?;
+    git_wrap::run_git_captured(None, "checkout", &git_args, true, None)?;
```

Comment:
ok, but I worry git_args can get long.

Reply:
There is pathspec-from-file which can read from a file or stdin, but it seems to exist only as of 2.27.